### DISCUSSION TITLE
AnalysisTemplate for GCP managed prometheus

### DISCUSSION THREAD

**Discussion by poonsalai** [argoproj/argo-workflows#12481] [Q&A] [answered]
Hi Team

In documentation (per below link) we have the example for Amazon Managed Prometheus but somehow we are unable to location any example for GCP managed prometheus..

https://argo-rollouts.readthedocs.io/en/stable/analysis/prometheus/#utilizing-amazon-managed-prometheus

When we are running a query onto the GCP console using PromQL it works fine and gives the result per below but it is failing when we run it using the AnalysisTemplate and we get 401 while doing it.. because we are not able to get any way to pass the authentication using AnalysisTemplate for GCP.

Our AnalysisTemplate looks like below :

<img width="949" alt="image" src="https://github.com/argoproj/argo-workflows/assets/105905102/7d6cf0cb-7cb7-4964-ac24-1e03a5eebc37">

When we run it we are getting 401 per below

<img width="907" alt="image" src="https://github.com/argoproj/argo-workflows/assets/105905102/39843d93-00fb-467f-a037-6e99001b8aad">

Could you please suggest any way how do we pass the authentication bit for GCP in AnalysisTemplate.. Below is the snippet for passing Auth in Amazon managed prometheus, need same way the snippet for GCP.

<img width="264" alt="image" src="https://github.com/argoproj/argo-workflows/assets/105905102/3fa5fd52-2dca-4ab0-9616-46c3a057078f">

Waiting for a reply that can help us moving forward, thanks a lot :)

Thanks

**Comment by agilgur5** [comment, marked as answer]
This is the Argo Workflows repo, it sounds like you want to post this in the [Argo Rollouts repo](https://github.com/argoproj/argo-rollouts/discussions).
